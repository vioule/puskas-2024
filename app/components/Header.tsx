"use client";
import Image from "next/image";
import { motion } from "framer-motion";

export default function Header() {
  const variants = {
    visible: {
      opacity: 1,
      transition: {
        staggerChildren: 0.25,
        delayChildren: 1,
        duration: 1,
      },
    },
    hidden: { opacity: 0 },
  };
  return (
    <header className="fixed top-0 w-full flex min-h-screen flex-col items-center justify-center p-24 bg-black">
      <motion.div
        className="flex flex-col max-w-[22.5rem]"
        data-testid="container"
        animate="visible"
        initial="hidden"
        variants={variants}
      >
        <motion.div className="relative h-20" variants={variants}>
          <Image src="/FIFA.svg" alt="FIFA logo" fill />
        </motion.div>
        <motion.span
          className="text-6xl text-gold tracking-[1.5rem] font-black font-roboto mt-4"
          variants={variants}
        >
          PUSKÃ€S
        </motion.span>
        <motion.span
          className="text-white text-2xl tracking-[0.5rem] text-right"
          variants={variants}
        >
          AWARD
        </motion.span>
        <motion.div
          className="flex justify-center items-center gap-1 mt-10"
          variants={variants}
        >
          <div className="w-[1px] h-4 bg-white animate-scroll origin-top"></div>
        </motion.div>
      </motion.div>
    </header>
  );
}
