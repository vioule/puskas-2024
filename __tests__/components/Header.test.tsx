import { render, screen, waitFor } from "@testing-library/react";
import Header from "@/app/components/Header";
import { motionValue } from "framer-motion";

test("render Header unchanged", () => {
  const { container } = render(
    <Header skewY={motionValue(10)} skewYNeg={motionValue(-10)} />
  );
  expect(container).toMatchSnapshot();
});

test("Header is visible after Framer motion animation", async () => {
  render(<Header skewY={motionValue(10)} skewYNeg={motionValue(-10)} />);
  const container = screen.getByTestId("container");
  expect(container).toHaveStyle("opacity: 0");
  expect(container.children[0]).toHaveStyle("opacity: 0");
  expect(container.children[1]).toHaveStyle("opacity: 0");
  expect(container.children[2]).toHaveStyle("opacity: 0");
  expect(container.children[2]).toHaveStyle("opacity: 0");
  await waitFor(
    () => {
      expect(container).toHaveStyle("opacity: 1");
      expect(container.children[0]).toHaveStyle("opacity: 1");
      expect(container.children[1]).toHaveStyle("opacity: 1");
      expect(container.children[2]).toHaveStyle("opacity: 1");
      expect(container.children[2]).toHaveStyle("opacity: 1");
    },
    { timeout: 2500 }
  );
});
