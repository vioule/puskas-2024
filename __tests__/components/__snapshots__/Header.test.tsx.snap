// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header unchanged 1`] = `
<div>
  <header
    class="flex min-h-screen flex-col items-center justify-center p-24 relative"
  >
    <div
      class="flex flex-col max-w-[22.5rem]"
      data-testid="container"
      style="opacity: 0;"
    >
      <div
        class="relative h-20"
        style="opacity: 0;"
      >
        <img
          alt="FIFA logo"
          data-nimg="fill"
          decoding="async"
          loading="lazy"
          src="/FIFA.svg"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
      </div>
      <span
        class="text-6xl text-gold tracking-[1.5rem] font-black font-roboto mt-4"
        style="opacity: 0;"
      >
        PUSKÃ€S
      </span>
      <span
        class="text-white text-2xl tracking-[0.5rem] text-right"
        style="opacity: 0;"
      >
        AWARD
      </span>
      <div
        class="flex justify-center items-center gap-1 mt-10"
        style="opacity: 0;"
      >
        <div
          class="w-[1px] h-4 bg-white animate-scroll origin-top"
        />
      </div>
    </div>
  </header>
</div>
`;
